Solutions/Azure Active Directory/Analytic Rules/MaliciousOAuthApp_PwnAuth.yaml

```diff
@@ -56,20 +77,26 @@ query: |
   | where LoggedByService =~ "Core Directory"
   | where Category =~ "ApplicationManagement"
   | where OperationName =~ "Add OAuth2PermissionGrant" or OperationName =~ "Add delegated permission grant"
-  | extend GrantAuthentication = tostring(TargetResources[0].displayName)
+        | mv-apply TargetResource = TargetResources on 
+            (
+                where TargetResource.type =~ "ServicePrincipal" and array_length(TargetResource.modifiedProperties) > 0 and isnotnull(TargetResource.displayName)
```

Comment:
@petebryan There were multiple ServicePrincipal items in the record but only one had non-null displayName and modifiedProperties. Can you validate if this modified condition looks ok to you?

Reply:
yes that looks correct in all instances